trellis/containers/test/test_transforms.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2022 Agtonomy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include "trellis/containers/transforms.hpp"
+
+using namespace trellis::core;
+
+static constexpr std::chrono::milliseconds kTransformHistoryWindow(1000);
+
+TEST(TrellisTransforms, HasTransformTestSuccessCase) {
+  time::EnableSimulatedClock();
+  time::SetSimulatedTime(time::TimePoint(std::chrono::milliseconds(5000)));  // initial time
+  trellis::containers::Transforms transforms;
+
+  // Initially we don't have this transform
+  ASSERT_FALSE(transforms.HasTransform("foo", "bar"));
+
+  trellis::containers::Transforms::RigidTransform transform;
+  transforms.UpdateTransform("foo", "bar", transform);
+  ASSERT_TRUE(transforms.HasTransform("foo", "bar"));
+}
+
+TEST(TrellisTransforms, RetrieveCorrectTransformGivenATime) {
+  time::EnableSimulatedClock();
+  auto now = time::TimePoint(std::chrono::milliseconds(1000));  // initial time
+
+  trellis::containers::Transforms transforms;
+  trellis::containers::Transforms::RigidTransform transform;
+  for (unsigned i = 0; i < 10U; ++i) {
+    transform.translation.x() = static_cast<double>(i + 1);
+    time::SetSimulatedTime(now);  // initial time
+    transforms.UpdateTransform("foo", "bar", transform);
+    now += std::chrono::milliseconds(50);  // 20 Hz update rate
+  }
+  // Now we have transforms with timestamps starting from 1000ms with 50ms increments
+
+  {
+    // We're back in time before the first transform by just within the validity window
```

Comment:
Simpler code without the validity window, nice! There are a few comments around the PR that still have the phrase "validity" in them FYI that we may wish to clean up.